client/app/components/group/_groupEditCreateForm/groupEditCreateForm.spec.js

```diff
@@ -86,6 +86,60 @@ describe("GroupEditCreateForm", () => {
       });
       expect(leafletMarker.lat).to.equal(99.99);
     });
+  });
+
+  describe("directive", () => {
+    let scope, $componentController, el, GroupService, $q, ngModel;
+
+    beforeEach(() => {
+      inject(($injector, $rootScope, $compile) => {
+        scope = $rootScope.$new();
+        $componentController = $injector.get("$componentController");
+        scope.$ctrl = $componentController("groupEditCreateForm", {}, {
+          data: {
+            id: 1
+          }
+        });
+        $q = $injector.get("$q");
+        GroupService = $injector.get("GroupService");
+        el = angular.element(
+        "<input type='text' name='name' ng-model='$ctrl.data.name' groupname-Validator/>"
+        );
+        $compile(el)(scope);
+      });
+      ngModel = el.controller("ngModel");
+    });
 
+    it("should reject if group name already exists", () => {
+      sinon.stub(GroupService, "listByGroupName").returns($q((resolve) => {
```

Comment:
I like the command chain here, seems slightly more readable than what I wrote in the other tests. Will use this style in future.